test/system/src/main/java/io/pravega/test/system/framework/services/ZookeeperDockerService.java

```diff
@@ -0,0 +1,101 @@
+import com.spotify.docker.client.messages.swarm.EndpointSpec;
+import com.spotify.docker.client.messages.swarm.NetworkAttachmentConfig;
+import com.spotify.docker.client.messages.swarm.PortConfig;
+import com.spotify.docker.client.messages.swarm.ResourceRequirements;
+import com.spotify.docker.client.messages.swarm.Resources;
+import com.spotify.docker.client.messages.swarm.Service;
+import com.spotify.docker.client.messages.swarm.ServiceMode;
+import com.spotify.docker.client.messages.swarm.ServiceSpec;
+import com.spotify.docker.client.messages.swarm.TaskSpec;
+import lombok.extern.slf4j.Slf4j;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+
+
+@Slf4j
+public class ZookeeperDockerService extends DockerBasedService {
+
+     private static final String ZK_IMAGE = "jplock/zookeeper:3.5.1-alpha";
+     private int instances = 1;
+     private double cpu = 1.0 * Math.pow(10.0, 9.0);
+     private long mem = 1024 * 1024 * 1024L;
+
+     public ZookeeperDockerService(String serviceName) {
+     super(serviceName);
+     }
+
+     @Override
+     public void stop() {
+         try {
+             Service.Criteria criteria = Service.Criteria.builder().serviceName(this.serviceName).build();
+             List<Service> serviceList = docker.listServices(criteria);
+             for (int i = 0; i < serviceList.size(); i++) {
+                 String serviceId = serviceList.get(i).id();
+                 docker.removeService(serviceId);
+             }
+         } catch (DockerException | InterruptedException e) {
+             log.error("unable to remove service {}", e);
+         }
+     }
+
+    @Override
+    public void clean() {
+    }
+
+    @Override
+    public void start(final boolean wait) {
+        try {
+            ServiceCreateResponse serviceCreateResponse = docker.createService(setServiceSpec());
+            if (wait) {
+                waitUntilServiceRunning().get(5, TimeUnit.MINUTES);
+            }
+           assertThat(serviceCreateResponse.id(), is(notNullValue()));
+        } catch (InterruptedException | DockerException | ExecutionException | TimeoutException e) {
+            log.error("unable to create service {}", e);
+        }
+    }
+
+    private ServiceSpec setServiceSpec() {
+
+        final TaskSpec taskSpec = TaskSpec
+                .builder()
+                .containerSpec(ContainerSpec.builder().image(ZK_IMAGE).command("/opt/zookeeper/bin/zkServer.sh", "start-foreground")
+                .healthcheck(ContainerConfig.Healthcheck.create(null,
+                        1000000000L, 1000000000L, 3)).build())
+                .resources(ResourceRequirements.builder()
+                        .limits(Resources.builder().memoryBytes(mem).nanoCpus((long) cpu).build())
```

Comment:
To Configuring CPUs in docker API we need to do the following:

* to configure 1.5 cpus to a docker container implies --cpu-period="100000" and --cpu-quota="150000"

* to configure 0.5 cpus to a docker container implies --cpu-period=100000 --cpu-quota=50000
i.e. This will translate to the following code:

```
Constant CPU_PERIOD= 100000L
float cpus = 1.5; // input by each service impl(zookeeperService, BookkeeperService etc...)
if should translate into 
HostConfig.builder().cpuShares(Math.round(cpus * CPU_PERIOD)).cpuPeriod(CPU_PERIOD)
```